src/cmdline/mod.rs

```diff
@@ -336,101 +287,88 @@ impl Cmdline {
     }
 }
 
+impl TryFrom<(usize, String)> for Cmdline {
+    type Error = ();
+
+    fn try_from(
+        (cmdline_capacity, cmdline_raw): (usize, String),
+    ) -> result::Result<Self, Self::Error> {
+        let (boot_args, init_args) = cmdline_raw.split_once("-- ").unwrap();
+
+        if boot_args.len() + init_args.len() + CMDLINE_NO_OF_AUXILIARY_CHARS > cmdline_capacity {
+            return Err(());
+        }
+
+        Ok(Cmdline {
+            boot_args: boot_args.trim().to_string(),
+            init_args: init_args.trim().to_string(),
+            cmdline_capacity,
+        })
+    }
+}
+
 impl From<Cmdline> for Vec<u8> {
     fn from(cmdline: Cmdline) -> Vec<u8> {
-        cmdline.line.into_bytes()
+        cmdline.as_str().to_string().into_bytes()
     }
 }
 
 impl PartialEq for Cmdline {
     fn eq(&self, other: &Self) -> bool {
-        self.line == other.line
+        self.as_str() == other.as_str()
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::ffi::CString;
+    use std::convert::TryFrom;
 
-    #[test]
-    fn test_insert_hello_world() {
-        let mut cl = Cmdline::new(100);
-        assert_eq!(cl.as_str(), "");
-        assert!(cl.insert("hello", "world").is_ok());
-        assert_eq!(cl.as_str(), "hello=world");
-
-        let s = CString::new(cl).expect("failed to create CString from Cmdline");
-        assert_eq!(s, CString::new("hello=world").unwrap());
-    }
+    const DEFAULT_CMDLINE_RAW : &str = "console=ttyS0 nomodules random.trust_cpu=on reboot=k panic=1 pci=off ip=:::::eth0:dhcp init=/bin/cat -- /etc/password";
```

Comment:
I don't think the *default* command line we test with needs to be this explicit. Several of these boot args bring no additional value - from a testing standpoint, all key-value args are the same.